File: bookstore_backend/src/main/java/com/example/bookstore_backend/service/CartService.java
package com.example.bookstore_backend.service;

import com.example.bookstore_backend.entity.Book;
import java.math.BigDecimal;
import java.util.List;

public interface CartService {
  List<Book> getCartItems(Integer user_id);

  void deleteCartItems(Integer user_id,Long book_id);

  void addCartItems(Integer user_id,Long book_id);

  void createOrder( Integer user_id,List<Long> book_id,
      List<Integer> amount, List<BigDecimal> price);

  boolean checkInventory(List<Long> book_id,
      List<Integer> amount);
}

File: bookstore_frontend/src/services/cartService.js
import {postRequest_v2} from "../utils/ajax";

export const getCartItems = (user_id, callback) => {
    const data = {user_id: user_id};
    const url = 'http://localhost:8080/getCartItems';
    console.log(data);
    postRequest_v2(url, data, callback);
};

export const deleteCartItems = (user_id, book_id, callback) => {
    const data = {user_id: user_id, book_id: book_id};
    const url = 'http://localhost:8080/deleteCartItems';
    console.log(data);
    postRequest_v2(url, data, callback);
};

export const addCartItems = (user_id, book_id, callback) => {
    const data = {user_id: user_id, book_id: book_id};
    console.log(user_id, book_id);
    const url = 'http://localhost:8080/addCartItems';
    console.log(data);
    postRequest_v2(url, data, callback);
};

export const createOrder = (user_id, book_id, amount, price, callback) => {
    const data = {user_id: user_id, book_id: book_id, amount: amount, price: price};
    console.log(user_id, book_id);
    const url = 'http://localhost:8080/createOrder';
    console.log(data);
    postRequest_v2(url, data, callback);
};

export const checkInventory = (book_id, amount, callback) => {
    const data = {book_id: book_id, amount: amount};
    const url = 'http://localhost:8080/checkInventory';
    postRequest_v2(url, data, callback);
};